[lib] Dependencies (save-pixels) not browser-compatible

The `save-pixels` package has dependencies (`contentstream`, at least) that aren't browser-compatible. Some of its dependencies appear to go out of their way to stick to plain JS, so I'm surprised to learn that. May need to investigate making a PR, or replacing it with another ndarray -> bytes implementation. 

Worth a look:

- https://github.com/eponymous-labs/browser-save-pixels
- https://www.npmjs.com/package/@nsfw-filter/save-pixels

Didn't look into `save-pixels` yet.

***

This affects the `@gltf-transform/lib` dependency; `@gltf-transform/core` and `@gltf-transform/extensions` do not depend on this package.

Turns out to be a lot of things. Workarounds seem to be bundler-specific, so I'd like to remove these dependencies where possible.

<details>

<summary>Log output</summary>

```
ERROR in ./node_modules/contentstream/index.js 18:11-26
Module not found: Error: Can't resolve 'util' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/contentstream'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/contentstream/node_modules/readable-stream/lib/_stream_readable.js 30:13-37
Module not found: Error: Can't resolve 'buffer' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/contentstream/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }
 @ ./node_modules/contentstream/node_modules/readable-stream/readable.js 2:10-63
 @ ./node_modules/contentstream/index.js 17:15-50
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/contentstream/node_modules/readable-stream/lib/_stream_readable.js 43:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/contentstream/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/contentstream/node_modules/readable-stream/readable.js 2:10-63
 @ ./node_modules/contentstream/index.js 17:15-50
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/contentstream/node_modules/readable-stream/lib/_stream_writable.js 29:13-37
Module not found: Error: Can't resolve 'buffer' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/contentstream/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }
 @ ./node_modules/contentstream/node_modules/readable-stream/readable.js 5:0-55
 @ ./node_modules/contentstream/index.js 17:15-50
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/contentstream/node_modules/readable-stream/lib/_stream_writable.js 40:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/contentstream/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/contentstream/node_modules/readable-stream/readable.js 5:0-55
 @ ./node_modules/contentstream/index.js 17:15-50
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/contentstream/node_modules/readable-stream/readable.js 1:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/contentstream/node_modules/readable-stream'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/contentstream/index.js 17:15-50
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/contentstream/node_modules/readable-stream/readable.js 10:2-36
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/contentstream/node_modules/readable-stream'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/contentstream/index.js 17:15-50
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/get-pixels/dom-pixels.js 3:20-35
Module not found: Error: Can't resolve 'path' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/get-pixels'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
	- install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "path": false }
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:372-398 1:7370-7371
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/gif-encoder/lib/GIFEncoder.js 11:13-30
Module not found: Error: Can't resolve 'assert' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/gif-encoder/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "assert": require.resolve("assert/") }'
	- install 'assert'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "assert": false }
 @ ./node_modules/save-pixels/save-pixels.js 4:17-39
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/gif-encoder/lib/GIFEncoder.js 14:11-26
Module not found: Error: Can't resolve 'util' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/gif-encoder/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
 @ ./node_modules/save-pixels/save-pixels.js 4:17-39
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/gif-encoder/node_modules/readable-stream/lib/_stream_readable.js 30:13-37
Module not found: Error: Can't resolve 'buffer' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/gif-encoder/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }
 @ ./node_modules/gif-encoder/node_modules/readable-stream/readable.js 1:10-63
 @ ./node_modules/gif-encoder/lib/GIFEncoder.js 13:21-47
 @ ./node_modules/save-pixels/save-pixels.js 4:17-39
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/gif-encoder/node_modules/readable-stream/lib/_stream_readable.js 43:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/gif-encoder/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/gif-encoder/node_modules/readable-stream/readable.js 1:10-63
 @ ./node_modules/gif-encoder/lib/GIFEncoder.js 13:21-47
 @ ./node_modules/save-pixels/save-pixels.js 4:17-39
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/gif-encoder/node_modules/readable-stream/lib/_stream_writable.js 29:13-37
Module not found: Error: Can't resolve 'buffer' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/gif-encoder/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }
 @ ./node_modules/gif-encoder/node_modules/readable-stream/readable.js 4:0-55
 @ ./node_modules/gif-encoder/lib/GIFEncoder.js 13:21-47
 @ ./node_modules/save-pixels/save-pixels.js 4:17-39
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/gif-encoder/node_modules/readable-stream/lib/_stream_writable.js 40:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/gif-encoder/node_modules/readable-stream/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/gif-encoder/node_modules/readable-stream/readable.js 4:0-55
 @ ./node_modules/gif-encoder/lib/GIFEncoder.js 13:21-47
 @ ./node_modules/save-pixels/save-pixels.js 4:17-39
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/gif-encoder/node_modules/readable-stream/readable.js 2:0-34
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/gif-encoder/node_modules/readable-stream'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/gif-encoder/lib/GIFEncoder.js 13:21-47
 @ ./node_modules/save-pixels/save-pixels.js 4:17-39
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/chunkstream.js 4:11-26
Module not found: Error: Can't resolve 'util' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
 @ ./node_modules/pngjs-nozlib/lib/parser-async.js 5:18-42
 @ ./node_modules/pngjs-nozlib/lib/png.js 5:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/chunkstream.js 5:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/pngjs-nozlib/lib/parser-async.js 5:18-42
 @ ./node_modules/pngjs-nozlib/lib/png.js 5:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/filter-parse-async.js 3:11-26
Module not found: Error: Can't resolve 'util' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
 @ ./node_modules/pngjs-nozlib/lib/parser-async.js 6:18-49
 @ ./node_modules/pngjs-nozlib/lib/png.js 5:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/packer-async.js 3:11-26
Module not found: Error: Can't resolve 'util' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
 @ ./node_modules/pngjs-nozlib/lib/png.js 6:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/packer-async.js 4:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/pngjs-nozlib/lib/png.js 6:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/packer-sync.js 4:11-26
Module not found: Error: Can't resolve 'zlib' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "zlib": require.resolve("browserify-zlib") }'
	- install 'browserify-zlib'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "zlib": false }
 @ ./node_modules/pngjs-nozlib/lib/png-sync.js 5:11-35
 @ ./node_modules/pngjs-nozlib/lib/png.js 7:14-35
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/packer.js 7:11-26
Module not found: Error: Can't resolve 'zlib' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "zlib": require.resolve("browserify-zlib") }'
	- install 'browserify-zlib'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "zlib": false }
 @ ./node_modules/pngjs-nozlib/lib/packer-async.js 6:13-32
 @ ./node_modules/pngjs-nozlib/lib/png.js 6:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/parser-async.js 3:11-26
Module not found: Error: Can't resolve 'util' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
 @ ./node_modules/pngjs-nozlib/lib/png.js 5:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/parser-async.js 4:11-26
Module not found: Error: Can't resolve 'zlib' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "zlib": require.resolve("browserify-zlib") }'
	- install 'browserify-zlib'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "zlib": false }
 @ ./node_modules/pngjs-nozlib/lib/png.js 5:13-38
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/parser-sync.js 4:11-26
Module not found: Error: Can't resolve 'zlib' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "zlib": require.resolve("browserify-zlib") }'
	- install 'browserify-zlib'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "zlib": false }
 @ ./node_modules/pngjs-nozlib/lib/png-sync.js 4:12-36
 @ ./node_modules/pngjs-nozlib/lib/png.js 7:14-35
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/png.js 3:11-26
Module not found: Error: Can't resolve 'util' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "util": require.resolve("util/") }'
	- install 'util'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "util": false }
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/pngjs-nozlib/lib/png.js 4:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/pngjs-nozlib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/save-pixels/save-pixels.js 6:10-37
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/string_decoder/index.js 22:13-37
Module not found: Error: Can't resolve 'buffer' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/string_decoder'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }
 @ ./node_modules/contentstream/node_modules/readable-stream/lib/_stream_readable.js 116:22-62 217:20-60
 @ ./node_modules/contentstream/node_modules/readable-stream/readable.js 2:10-63
 @ ./node_modules/contentstream/index.js 17:15-50
 @ ./node_modules/save-pixels/save-pixels.js 3:20-44
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19

ERROR in ./node_modules/through/index.js 1:13-30
Module not found: Error: Can't resolve 'stream' in '/Users/donmccurdy/Projects/glTF-Report/node_modules/through'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }
 @ ./node_modules/save-pixels/save-pixels.js 9:14-32
 @ ./node_modules/@gltf-transform/lib/dist/lib.modern.js 1:398-425 1:7548-7549
 @ ./src/services/runner.ts 3:0-29
 @ ./src/Editor.svelte 25:0-46 134:5-14 165:2-11
 @ ./src/App.svelte 24:0-37 52:14-20 154:44-50
 @ ./src/main.ts 2:0-31 3:16-19
```

</details>

Problems:


- `save-pixels` > `contentstream` > ...
    - `util`
    - `buffer`
    - `stream`
- `save-pixels` > `through` > ...
    - `stream`
- `save-pixels` > `gif-encoder`
    - `assert`
    - `util`
    - `buffer`
    - `stream`
- `save-pixels` > `pngjs-nozlib`
    - `util`
    - `stream`
    - `zlib`
- `get-pixels` > ...
    - `path`

Fallbacks available:

- `util`: 34kb
- `stream-browserify`: 12kb
- `browserify-zlib`: 15kb
- ...

--- a/packages/lib/package.json
+++ b/packages/lib/package.json
@@ -24,11 +24,10 @@
     "@gltf-transform/core": "^0.10.1",
     "@gltf-transform/extensions": "^0.10.1",
     "geo-ambient-occlusion": "^3.0.2",
-    "get-pixels": "^3.3.2",
     "gl-matrix": "^3.3.0",
     "ndarray": "^1.0.19",
-    "regl": "^2.1.0",
-    "save-pixels": "^2.3.4"
+    "ndarray-pixels": "^0.1.2",
+    "regl": "^2.1.0"
   },
   "files": [
     "dist/",

--- a/packages/lib/src/utils.ts
+++ b/packages/lib/src/utils.ts
@@ -1,7 +1,6 @@
-import getPixels from 'get-pixels';
 import ndarray from 'ndarray';
-import savePixels from 'save-pixels';
-import { BufferUtils, Primitive, Texture } from '@gltf-transform/core';
+import { getPixels, savePixels } from 'ndarray-pixels';
+import { Primitive, Texture } from '@gltf-transform/core';
 
 /** Maps pixels from source to target textures, with a per-pixel callback. */
 export async function rewriteTexture(
@@ -11,14 +10,7 @@ export async function rewriteTexture(
 
 	if (!source) return null;
 
-	const pixels: ndarray = await new Promise((resolve, reject) => {
-		// eslint-disable-next-line @typescript-eslint/ban-types
-		(getPixels as unknown as Function)(
-			Buffer.from(source.getImage()),
-			source.getMimeType(),
-			(err, pixels) => err ? reject(err) : resolve(pixels)
-		);
-	});
+	const pixels = await getPixels(new Uint8Array(source.getImage()), source.getMimeType());
 
 	for(let i = 0; i < pixels.shape[0]; ++i) {
 		for(let j = 0; j < pixels.shape[1]; ++j) {
@@ -26,14 +18,7 @@ export async function rewriteTexture(
 		}
 	}
 
-	const image: ArrayBuffer = await new Promise((resolve, reject) => {
-		const chunks: Buffer[] = [];
-		savePixels(pixels, 'png')
-			.on('data', (d) => chunks.push(d))
-			.on('end', () => resolve(BufferUtils.trim(Buffer.concat(chunks))))
-			.on('error', (e) => reject(e));
-	});
-
+	const image = (await savePixels(pixels, 'image/png')).buffer;
 	return target.setImage(image).setMimeType('image/png');
 }
 

--- a/packages/lib/test/metal-rough.test.ts
+++ b/packages/lib/test/metal-rough.test.ts
@@ -1,9 +1,9 @@
 require('source-map-support').install();
 
 import ndarray from 'ndarray';
-import savePixels from 'save-pixels';
+import { savePixels } from 'ndarray-pixels';
 import test from 'tape';
-import { BufferUtils, Document } from '@gltf-transform/core';
+import { Document } from '@gltf-transform/core';
 import { IOR, MaterialsIOR, MaterialsPBRSpecularGlossiness, MaterialsSpecular, Specular } from '@gltf-transform/extensions';
 import { metalRough } from '../';
 
@@ -38,13 +38,7 @@ const ROUGH = ndarray(new Uint8Array([
 ]), [1, 4, 4]);
 
 async function ndarrayToImage (pixels): Promise<ArrayBuffer> {
-	return new Promise((resolve, reject) => {
-		const chunks = [];
-		savePixels(pixels, 'png')
-			.on('data', (d) => chunks.push(d))
-			.on('end', () => resolve(BufferUtils.trim(Buffer.concat(chunks))))
-			.on('error', (e) => reject(e));
-	});
+	return (await savePixels(pixels, 'image/png')).buffer;
 }
 
 test('@gltf-transform/lib::metalRough | textures', async t => {

